docs/org/reference/language/functions.org

```diff
@@ -37,13 +37,52 @@ Function declarations can depend on each other recursively. In the following exa
 #+begin_example
 open import Stdlib.Prelude;
 
-odd : Nat → Bool;
-
-even : Nat → Bool;
+odd : Nat -> Bool;
+even : Nat -> Bool;
 
 odd zero := false;
 odd (suc n) := even n;
 
 even zero := true;
 even (suc n) := odd n;
 #+end_example
+
+** Anonymous functions
+
+Anonymous functions, or /lambdas/, are introduced with the syntax:
+
+#+begin_example
+\{| pat1 .. patN_1 := clause1
+  | ..
+  | pat1 .. patN_M := clauseM}
+#+end_example
+
+The first pipe =|= is optional. Instead of =\= one can also use =λ=.
+
+An anonymous function just lists all clauses of a function without
+naming it. Any function declaration can be converted to use anonymous
+functions:
+
+#+begin_example
+open import Stdlib.Prelude;
+
+odd : Nat -> Bool;
+even : Nat -> Bool;
+
+odd := \{
+  zero := false;
+  (suc n) := even n;
+};
+
+even := \{
+  zero := true;
+  (suc n) := odd n;
+};
```

Comment:
Replace this by:

```
  odd := \ {
      | zero := false
      | (suc n) := even n
    };

  even := \ {
      | zero := true
      | (suc n) := odd n
    };
```

Reply:
Right, forgot about the pipes again

Reply:
It's fine; we'll address the main issue here: 
- #1839

Reply:
Done